### PR TITLE
[Website][CI] Build Website without Warnings + Add Workflow for Building Website 

### PR DESCRIPTION
**Pull request by barry-jin**
## Description ##

1. Fix some bugs and documentation issues to build website without any warnings
2. Add workflow for building website

TODO: 

- [ ] Include model_zoo documentation for website build
- [ ] Deploy website on github.io or S3

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

cc @dmlc/gluon-nlp-team

### CODE REVIEW

File: .github/workflows/buildwebsite.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+          architecture: x64
+
+      - name: Install Other Dependencies
+        run: |
+          python -m pip install --user --quiet --upgrade pip
+          python3 -m pip install setuptools pytest pytest-cov contextvars
```
Comment (leezu) [inline .github/workflows/buildwebsite.yml:30]: contextvars is not needed on python 3.7. Why do we need pytest for the website build?

File: .github/workflows/buildwebsite.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: continuous build - docs
+
+on: [push, pull_request]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  unittest-gpu:
```
Comment (leezu) [inline .github/workflows/buildwebsite.yml:10]:
```suggestion
  website-build:
```

File: scripts/conversion_toolkits/README.md [hunk 3]
```diff
@@ -16,9 +16,9 @@ For RoBERTa XLM-R and BART model, we rely on the master version of [fairseq](htt
 
 ## Convert all models
 
-``bash
+```bash
 bash convert_all.sh
-``
+```
```
Comment (szha) [inline scripts/conversion_toolkits/README.md:21]: this has been removed. let's remove this section

File: tools/batch/batch_states/compile_notebook.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import os
+import argparse
+import sys
+import re
+
+HOME_PATH = os.path.expandvars('$HOME') + "/work/gluon-nlp"
```
Comment (szha) [inline tools/batch/batch_states/compile_notebook.py:6]: use [`os.path.join`](https://docs.python.org/3/library/os.path.html#os.path.join) for portability.

File: tools/batch/batch_states/compile_notebook.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+class DeployWebsiteDriver(object):
+    def __init__(self, args):
+        self.md_file_list = []
+        if args.pr_number != '0':
+            self.pr_number = 'PR#' + args.pr_number
+        else: 
+            self.pr_number = 'push'
```
Comment (szha) [inline tools/batch/batch_states/compile_notebook.py:14]: is there a build number similar to jenkins?

File: tools/batch/batch_states/compile_notebook.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        md_file_list = []
+        try:
+            for file in os.listdir(md_file_path):
+                if os.path.isfile(os.path.join(md_file_path, file)):
+                    if file.endswith('.md'):
+                        self.md_file_list.append(os.path.join(md_file_path, file))
+                else:
+                    self.find_md_file(os.path.join(md_file_path, file))
+        except OSError:
+            print("[ERROR] %s does not exist" % md_file_path)
```
Comment (szha) [inline tools/batch/batch_states/compile_notebook.py:28]: use string.format in favor of `%` op.

**Review by szha — COMMENTED** [review]
what's the motivation to run shell commands in a python script?

File: tools/batch/batch_states/compile_notebook.py (hunk 5 above)
Comment (barry-jin) [reply tools/batch/batch_states/compile_notebook.py:14]: There is `github.run_number` for each run of a particular workflow.

**Comment by barry-jin** [comment]
> what's the motivation to run shell commands in a python script?

For me, I felt it more convenient to run and debug this workflow in python script rather than in shell script.

File: tools/batch/batch_states/compile_notebook.py (hunk 5 above)
Comment (szha) [reply tools/batch/batch_states/compile_notebook.py:14]: let's make sure it's in the path.

File: tools/batch/batch_states/compile_notebooks.sh [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            --command "python3 -m pip install --quiet nbformat notedown jupyter_client ipykernel && \
+                       python3 /gluon-nlp/docs/md2ipynb.py ${MDFILE}" 2>&1 | tee $LOGNAME >/dev/null
+
+    BATCH_EXIT_CODE=$?
+
+    head -100 $LOGNAME | grep -oP -m 1 'jobId: \K(.*)' > $JOBIDLOG
+
+    JOBID=$(cat "$JOBIDLOG")
+
+    python3 tools/batch/wait-job.py --job-id $JOBID
```
Comment (szha) [inline tools/batch/batch_states/compile_notebooks.sh:37]: this will make all the notebooks run sequentially on Batch, which defeats the purpose of using a job management system

instead, the script should submit all notebook jobs, collect the job ids, and then use a separate loop to wait on them.

Comment (barry-jin) [reply tools/batch/batch_states/compile_notebooks.sh:37]: In fact, I was using `&` to parallelize for-loop, so that these jobs can be run simultaneously.

Comment (szha) [reply tools/batch/batch_states/compile_notebooks.sh:37]: ah ok, totally missed it on line 51. thanks for clarifying.

File: .github/workflows/buildwebsite.yml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+          python-version: 3.7
+          architecture: x64
+
+      - name: Install Other Dependencies
+        run: |
+          python -m pip install --quiet --upgrade pip
+          python -m pip install --upgrade cython
+          python -m pip install --pre "mxnet>=2.0.0b20200802" -f https://dist.mxnet.io/python
+          python -m pip install --quiet -e .[extras]
+          python -m pip install setuptools ipython sphinx>=1.5.5 sphinx-gallery nbsphinx notedown sphinx_rtd_theme mxtheme sphinx-autodoc-typehints matplotlib Image recommonmark jupyter_client ipykernel
```
Comment (szha) [inline .github/workflows/buildwebsite.yml:33]: what are the sphinx-autodoc-typehints, Image, jupyter_client for? do we need to freeze their versions?

Comment (barry-jin) [reply .github/workflows/buildwebsite.yml:33]: `Image` is a package related to image and video resizing and cropping on website, which is suggested in [docs/README](https://github.com/dmlc/gluon-nlp/blob/v0.10.x/docs/README.txt). Some other dependencies like sphinx-autodoc-typehints, jupyter_client are required for compiling notebooks locally. Since, we are compiling notebooks on AWS Batch, I will remove these dependencies.

**Review by szha — APPROVED** [review]
LGTM. Thanks for setting up the website build!
